api/app/models/concerns/spree/webhooks/has_webhooks.rb

```diff
@@ -0,0 +1,47 @@
+module Spree
+  module Webhooks
+    module HasWebhooks
+      extend ActiveSupport::Concern
+
+      included do
+        after_create_commit proc { queue_webhooks_requests!(event_name(:create)) },
+          unless: proc { migrating? || webhooks_descendant? }
+        after_destroy_commit proc { queue_webhooks_requests!(event_name(:destroy)) },
+          unless: proc { migrating? || webhooks_descendant? }
+        after_update_commit proc { queue_webhooks_requests!(event_name(:update)) },
+          unless: proc { migrating? || webhooks_descendant? }
+
+        def queue_webhooks_requests!(event)
+          unless payload.blank?
+            Spree::Webhooks::Endpoints::QueueRequests.call(event: event, payload: payload) 
+          end
+        end
+      end
+
+      private
+
+      def event_name(operation)
+        "#{self.class.name.demodulize.tableize.singularize}.#{operation}"
+      end
+
+      def webhooks_descendant?
+        self.class.module_parents.include?(Spree::Webhooks)
+      end
+
+      def payload
+        resource_serializer.new(self).serializable_hash
+      rescue NameError, NoMethodError
+        {}
+      end
+
+      def resource_serializer
+        demodulized_class_name = self.class.to_s.demodulize
+        "Spree::Api::V2::Platform::#{demodulized_class_name}Serializer".constantize
+      end
+
+      def migrating?
```

Comment:
@sebastian-palma why do we need this?

Reply:
Because [this](https://github.com/spree/spree/blob/52a440857695acc5d72945b2c4d27eabfc3af3cb/core/db/migrate/20140309033438_create_store_from_preferences.rb#L19:L39) migration creates a store, doing so it executes the callback for queing a webhook request, and the [store serializer](https://github.com/spree/spree/blob/52a440857695acc5d72945b2c4d27eabfc3af3cb/api/app/serializers/spree/api/v2/platform/store_serializer.rb#L8) uses the store menus. All that happens before the `spree_menus` table is created which raises an error and the test app can't be created.

As I see, there are some migrations that create records, all of them would be executing the callbacks that way. For that case I think we don't need this code to be executed.

Reply:
@sebastian-palma that migration doesn't exist anymore, we don't need this anymore